src/QuickMarcEditor/QuickMarcEditor.js

```diff
@@ -80,15 +83,42 @@ const QuickMarcEditor = ({
     ? pristine || submitting
     : submitting;
 
-  const confirmSubmit = useCallback((e) => {
+  const redirectToVersion = useCallback((updatedVersion) => {
+    history.push({
+      search: `?relatedRecordVersion=${updatedVersion}`,
+    });
+  }, [history]);
+
+  const handleSubmitResponse = useCallback((updatedRecord, shouldClose) => {
+    if (!updatedRecord?.version) {
```

Comment:
It feels like there are many variables that are very similar in naming (`shouldClose`, `shouldCloseEditor`, `isKeepEditing`)  and it's hard to understand their purpose.

I suggest the following approach to simplify it:
1. Create a new variable `continueAfterSave` with `useRef` and set default value to `false`
2. In `confirmSubmit` take the `isKeepEditing` value and update `continueAfterSave.current = isKeepEditing`.
Because we used `useRef` instead of `useState` it means that changes to this value will be synchronous so we don't have to pass it as arguments anywhere.
3. In `confirmSubmit` we can now simply call `handleSubmit(e).then((updatedRecord) => handleSubmitResponse(updatedRecord));`
4. `handleSubmitResponse` can now be simplified to
```
const handleSubmitResponse = useCallback((updatedRecord) => {
    if (continueAfterSave.current) {
      redirectToVersion(updatedRecord.version);

      return;
    }

    onClose();
  }, [redirectToVersion, onClose]);
```

Reply:
That is fire! Great solution.

Yeah, regarding the this piece of code in `handleSubmitResponse`:
```
    if (!updatedRecord?.version) {
      reset();
      continueAfterSave.current = false;

      return;
    }
```
There are some fields that user cannot delete and in those cases `handleSubmit` returns `undefined`. So we need to restore deleted fields in frontend and reset `continueAfterSave` back to it's initial value.

Reply:
@odoniyor I see your point about when there's a validation error, I just don't think we should reset the whole form. This will be a problem if a user had made a lot of changes and made one mistake so validation failed. But because of reset instead of just fixing one mistake - all their changes were lost